rcl_action/test/rcl_action/test_action_communication.cpp

```diff
@@ -57,313 +65,1259 @@ class CLASSNAME (TestActionCommunication, RMW_IMPLEMENTATION) : public ::testing
   }
 
   void init_test_uuid0(uint8_t * uuid)
   {
-    for (uint8_t i = 0; i < 16; ++i) {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
       uuid[i] = i;
     }
   }
 
   void init_test_uuid1(uint8_t * uuid)
   {
-    for (uint8_t i = 0; i < 16; ++i) {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
       uuid[i] = 15 - i;
     }
   }
 
+  rcl_action_client_t action_client;
   rcl_action_server_t action_server;
   rcl_node_t node;
   rcl_clock_t clock;
 };  // class TestActionCommunication
 
-TEST_F(CLASSNAME(TestActionCommunication, RMW_IMPLEMENTATION), test_take_goal_request)
+TEST_F(CLASSNAME(TestActionCommunication, RMW_IMPLEMENTATION), test_valid_goal_comm)
 {
-  test_msgs__action__Fibonacci_Goal_Request goal_request;
-  test_msgs__action__Fibonacci_Goal_Request__init(&goal_request);
+  test_msgs__action__Fibonacci_Goal_Request outgoing_goal_request;
+  test_msgs__action__Fibonacci_Goal_Request incoming_goal_request;
+  test_msgs__action__Fibonacci_Goal_Response outgoing_goal_response;
+  test_msgs__action__Fibonacci_Goal_Response incoming_goal_response;
+  test_msgs__action__Fibonacci_Goal_Request__init(&outgoing_goal_request);
+  test_msgs__action__Fibonacci_Goal_Request__init(&incoming_goal_request);
+  test_msgs__action__Fibonacci_Goal_Response__init(&outgoing_goal_response);
+  test_msgs__action__Fibonacci_Goal_Response__init(&incoming_goal_response);
+
+  // Initialize goal request
+  init_test_uuid0(outgoing_goal_request.uuid);
+  outgoing_goal_request.order = 10;
+
+  // Send goal request with valid arguments
+  int64_t sequence_number;
+  rcl_ret_t ret = rcl_action_send_goal_request(
+    &this->action_client, &outgoing_goal_request, &sequence_number);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+
+  size_t num_subscriptions = 0;
+  size_t num_guard_conditions = 0;
+  size_t num_timers = 0;
+  size_t num_clients = 0;
+  size_t num_services = 0;
+
+  ret = rcl_action_server_wait_set_get_num_entities(
+    &this->action_server,
+    &num_subscriptions,
+    &num_guard_conditions,
+    &num_timers,
+    &num_clients,
+    &num_services);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait_set_init(
+    &wait_set,
+    num_subscriptions,     // number_of_subscriptions
+    num_guard_conditions,  // number_of_guard_conditions
+    num_timers,            // number_of_timers
+    num_clients,           // number_of_clients
+    num_services,          // number_of_services
+    rcl_get_default_allocator());
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_wait_set_add_action_server(&wait_set, &this->action_server, NULL);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  bool is_goal_request_ready(false);
+  bool is_cancel_request_ready(false);
+  bool is_result_request_ready(false);
```

Comment:
@apojomovsky here and everywhere else it applies, consider using `{}` instead.

Reply:
done!